design-proposals/mass_machine_type_transition.md

```diff
@@ -0,0 +1,208 @@
+﻿# Overview
+This proposal discusses a method to automate updating the machine type of VMs to the latest machine type version, if the VM has a machine type that is no longer compatible. Currently we can support manually changing the machine type of individual VMs, but we need an automated process that will be able to change the machine types of multiple (e.g. thousands of) VMs, limiting workload interruptions and preventing the user from having to manually update the machine type of every VM with a machine type version that is no longer supported. Centos 9 will maintain compatibility with all `pc-q35-rhel8.x.x` machine types throughout its lifecycle. However, with the transition to Centos 10, compatibility with machine types prior to `pc-q35-rhel9.0.0` will not be maintained. This means each workload must be updated to use a `pc-q35-rhel9.x.x` machine type.
```

Comment:
Since `pc-i440fx-rhel7.6.0` is listed as deprecated here, when encountering a machine type of the alias `pc` should it be handled the same way as `q35` and do we want to replace it with a `q35` machine type entirely?

Reply:
As written in QEMU's docs [1]:
> These old machine types are quite neglected nowadays and thus might have various pitfalls with regards to live migration. Use a newer machine type instead.

IOW, this machine type is deprecated. However, it is a different machine type than q35.

I think there might be a high probability that whoever chose this machine type might be interested in it specifically.
WDYT about avoiding changing it automatically to q35, but instead adding a warning label?

[1] https://www.qemu.org/docs/master/about/deprecated.html

Reply:
It's deprecated and was scheduled to be removed in EL 9 before being punted to EL 10 AFAIK and prompted the OpenStack tooling that I previously worked on.

Totally agree that we shouldn't migrate VMs automatically as the `pc` chipset differs greatly to `q35` and could require manual changes to the VM before it's bootable.

The following OpenStack docs cover some of this:

https://docs.openstack.org/nova/latest/admin/hw-machine-type.html